dashboard/test/controllers/report_abuse_controller_test.rb

```diff
@@ -0,0 +1,183 @@
+require 'test_helper'
+require 'cdo/aws/s3'
+
+class ReportAbuseControllerTest < ActionController::TestCase
+  include Devise::Test::ControllerHelpers
+
+  setup do
+    # channels
+    # Use a fake storage ID, because if you use the project_storage factory,
+    # everything breaks.
+    storage_id = 1
+    @projects = Projects.new(storage_id)
+    @channel_id = @projects.create({}, ip: '10.0.0.1')
+    @controller.stubs(:get_storage_id).returns(@storage_id)
+
+    # files
+    AWS::S3.stubs :create_client # Don't actually talk to S3
+    FileBucket.any_instance.stubs(:list).returns([{filename: 'test.file'}])
+  end
+
+  teardown do
+    @controller.unstub(:get_storage_id)
+
+    FileBucket.any_instance.unstub(:list)
+    FileBucket.any_instance.unstub(:get_abuse_score)
+  end
+
+  # channels
+
+  test "signed out user does not update abuse score" do
+    assert_equal 0, @controller.update_channel_abuse_score(@channel_id)
+  end
+
+  test "signed in update abuse score" do
+    DCDO.stubs(:get).with('restrict-abuse-reporting-to-verified', true).returns(false)
```

Comment:
can you please make this test case name more specific, like "signed in user can update abuse score without dcdo flag enabled", and also add tests covering the 2 cases for when the flag is enabled:
* signed in user cannot update
* signed in verified teacher can update

there is a factory method called `:authorized_teacher` which can help with this.